### PR TITLE
#150 Add connections page

### PR DESCRIPTION
**Pull request by nevendyulgerov**
I added a settings group to the navbar. It contains a single nav link to the connections page. I also removed a couple of obsolete tests for `ConnectionView` component and added several for the `Shell` component.

### CODE REVIEW

File: apps/web/src/components/connection/connectionView.tsx [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
-import { useDisclosure, useViewportSize } from "@mantine/hooks";
-import { TbChevronDown, TbChevronUp, TbPlus } from "react-icons/tb";
+import { TbPlus } from "react-icons/tb";
 import { useConnectionConfig } from "../../providers/connectionConfig/hooks";
 import ConnectionInfo from "./connectionInfo";
 
 const ConnectionView = () => {
-    const [showConnectionList, { toggle: toggleConnectionList }] =
-        useDisclosure(true);
     const { showConnectionModal } = useConnectionConfig();
```
Comment (dandheedge) [inline apps/web/src/components/connection/connectionView.tsx:17]: I think the `showConnectionModal` and `listConnections` could be on the same line

```suggestion
    const { showConnectionModal, listConnections } = useConnectionConfig();
```

File: apps/web/src/components/layout/shell.tsx
Comment (dandheedge) [inline apps/web/src/components/layout/shell.tsx:1]: https://github.com/cartesi/rollups-explorer/blob/75e74e818a72c9a3155779618d124ccc42b22f55/apps/web/src/components/layout/shell.tsx#L56

We can enable the transition by using this

```
const { colorScheme, toggleColorScheme } = useMantineColorScheme({
        keepTransitions: true,
    });
```

Comment (nevendyulgerov) [reply apps/web/src/components/layout/shell.tsx:1]: Nice, it's much better with transitions after this change :)

File: apps/web/src/components/connection/connectionInfo.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ const ConnectionInfo: FC<ConnectionInfoProps> = ({ connection }) => {
     const { removeConnection } = useConnectionConfig();
     const theme = useMantineTheme();
     return (
-        <Card withBorder py="lg" my="md" radius="sm" shadow="sm">
+        <Card withBorder py="lg" mt="md" radius="sm" shadow="sm">
```
[the remark this answers is not in the document]
Comment (nevendyulgerov) [reply apps/web/src/components/connection/connectionInfo.tsx:24]: I assume we are good with this `mt` setting or you suggest that I stick to `my`?

**Review by brunomenezes — COMMENTED** [review]
Hey @nevendyulgerov, I had a run with the page. I think the `ConnectionView` could be updated from a `Stack` approach to using a `Grid` to better utilize the new space; before the space was smaller, so it was the only sensible choice. Thinking of breaking points, it could have 1 column as a **base**, and anything from **md** and above goes with two columns. What do you think?

File: apps/web/src/components/connection/connectionInfo.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dandheedge) [reply apps/web/src/components/connection/connectionInfo.tsx:24]: After reviewing the UI again, I find it more balanced to use mt on this component. Sorry for the confusion, @nevendyulgerov.

**Review by brunomenezes — APPROVED** [review]
@nevendyulgerov, I rebased and pushed a small change to the menu for the small screen; it was not closing the sub-menu when clicking.

**Pull request merged**